core/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java

```diff
@@ -519,7 +519,14 @@ public Settings archiveUnknownOrInvalidSettings(
             try {
                 Setting<?> setting = get(entry.getKey());
                 if (setting != null) {
-                    setting.get(settings);
+                    try {
```

Comment:
Nit: I think we need to renamed the method + docs - it now also rejects mandatory and invalid settings.